command/agent/config.go

```diff
@@ -701,6 +701,11 @@ type ServerConfig struct {
 
 	// JobTrackedVersions is the number of historic job versions that are kept.
 	JobTrackedVersions *int `hcl:"job_tracked_versions"`
+
+	// OIDCIssuer if set enables OIDC Discovery and uses this value as the
+	// issuer. Third parties such as AWS IAM OIDC Provider expect the issuer to
+	// be a publically accessible HTTPS URL signed by a trusted well-known CA.
+	OIDCIssuer string `hcl:"oidc_issuer"`
```

Comment:
If there's a possibility we add more OIDC configs based on needs, would it make sense to encapsulate them in a block instead of a top-level string?

And we don't update this as much as we should, but could we validate the shape of this value in the `nomad config validate` command? It seems like there's some specific expectations to how it should look like.
https://github.com/hashicorp/nomad/blob/0410b8acea5e6ac1d032f6b98ce1b893f0230d92/command/agent/command.go#L311

Reply:
> If there's a possibility we add more OIDC configs based on needs

I brainstormed in Slack a bit and just didn't feel confident that I could come up with a new block that was any more likely to stand the test of time than this parameter. :disappointed:

`oidc { issuer = "..." }` is the obvious choice, but since Nomad has an OIDC Auth Method I thought that would be too confusing.

`workload_identity { oidc_issuer = "..." }` is wordy but at least is specific to this use case. That being said I think it's even odds we reuse this infrastructure for Node Identities (so a Node gets a JWT that it can perform operations with)... at which point we're full of regret for sticking this under `workload_identity`.

I also don't have an idea of what further parameters we'll need. We're probably stuck with `RS256` as the only reasonable algorithm. `jwks_uri` is built in a well-defined way on top of `issuer`, so I don't see why we would need to configure that distinctly.

So in the absence of a clear winner, I think putting it in `server` at least makes it clear where it's actually used (server-only). A feature that top-level blocks would lack.

> And we don't update this as much as we should, but could we validate the shape of this value in the nomad config validate command? 

Thanks for reminding me! Added. Kind of awkward we don't reuse that during agent startup, but it's better than nothing.